streaming/src/main/scala/org/apache/spark/streaming/Checkpoint.scala

```diff
@@ -102,7 +102,7 @@ class Checkpoint(ssc: StreamingContext, val checkpointTime: Time)
 private[streaming]
 object Checkpoint extends Logging {
   val PREFIX = "checkpoint-"
-  val REGEX = (PREFIX + """([\d]+)([\w\.]*)""").r
```

Comment:
Could you add a unit test case to `streaming/src/test/scala/org/apache/spark/streaming/CheckpointSuite.scala`?
Please use a test case name, `SPARK-28912 Fixed MatchError in getCheckpointFiles`. That should be fail without your patch.